tests/autoyast/installation.pm

```diff
@@ -215,6 +216,10 @@ sub run {
             $num_errors++;
         }
         elsif (match_has_tag('warning-pop-up')) {
+            # in order to avoid to match several times the same already processed warning
+            my @warnings_skip = grep { match_has_tag($_) } @processed_warnings;
+            next if scalar @warnings_skip > 0;
```

Comment:
This might break some scenarios, as we might get multiple warning. To make it work as per your proposal we need to make sure that we add unique tags we have on top of `warning-pop-up` and don't process them twice.

Reply:
could you help me understand how this could break other scenario? Basically this code as it could be more warning, it still match for `warning-pop-up` but if the specific warning in the list expected coming from test_data was already process, we don't need a timeout, we just skip it, which fix the issue. It doesn't extend the current functionality though, so if there are more than one error of the same type it will not match, but I was not planning to fix that, we need specific example where we use test_data and there will me multiple matches, I didn't find one.
For other warnings not coming from expected list it would continue doing what it was doing because it will have nothing to skip.

Reply:
So my concern is for the following case. We have multiple warnings to process.
First one is processed properly and `handle_warnings` is called. With your change, second time we get needle with `warning-pop-up` you will skip processing. This fixes the issue for single scenario only, when we expect single warning only.

Also, I see that `expected_warnings` are using test data structure, so it's definitely not used in every single case.
What we could do is to at least check recent matches of `warning-pop-up` on osd and o3. After that we might know f can use your approach. For me it doesn't look safe

Reply:
> warning-pop-up

Second time will not skip, because current screen will not match that specific warning due to it will be a different warning so `scalar @warnings_skip` will be 0.

Reply:
Ok, I see actually now you don't remove tags from `@expected_warnings` array. So case we need to take care of is if we still have scenarios where `@expected_warnings` is not set properly, but some warnings were processed, as now you make this check more strict. 
Also, as we don't use `warnings_skip` anywhere, I would propose to write this condition in single line. But that's minor.

Reply:
> Ok, I see actually now you don't remove tags from `@expected_warnings` array. So case we need to take care of is if we still have scenarios where `@expected_warnings` is not set properly, but some warnings were processed, as now you make this check more strict.
> Also, as we don't use `warnings_skip` anywhere, I would propose to write this condition in single line. But that's minor.

I searched for `expected_warnings` and we only set it in this scenario for now.
Good idea, no need for assignation. Fixed.